### PR TITLE
refactor(material/chips): change mat-chip to use MDC's token API

### PR DESCRIPTION
**Pull request by wagnermaciel**

### CODE REVIEW

File: src/material/chips/_chips-theme.scss [hunk 1]
```diff
@@ -1,101 +1,71 @@
...
+@mixin _colored-chip($config, $palette-name) {
+  $palette: map.get($config, $palette-name);
   $background: theming.get-color-from-palette($palette);
   $foreground: theming.get-color-from-palette($palette, default-contrast);
 
   &.mat-mdc-chip-selected,
   &.mat-mdc-chip-highlighted {
-    @include _chip-variant($background, $foreground);
+    @include mdc-chip-theme.theme((
+      elevated-container-color: $background,
```
Comment (crisbeto) [inline src/material/chips/_chips-theme.scss:15]: I think that the color variants are supposed to be generated in the tokens as well. It came up during the checkbox code review https://github.com/angular/components/pull/26744#discussion_r1130720783

File: src/material/chips/chip.scss [hunk 2]
```diff
@@ -48,62 +70,19 @@
...
       mdc-chip-theme.$trailing-action-trailing-padding
     );
-    @include mdc-chip-theme.graphic-size(mdc-chip-theme.$avatar-size);
-    @include mdc-chip-theme.icon-size(mdc-chip-theme.$avatar-size);
   }
 }
 
 @include mdc-helpers.disable-mdc-fallback-declarations {
-  @include mdc-chip.static-styles();
+  // @include mdc-chip.static-styles();
```
Comment (crisbeto) [inline src/material/chips/chip.scss:77]: Seems like a leftover

File: src/material/chips/chip.scss [hunk 3]
```diff
@@ -48,62 +70,19 @@
...
   @include _missing-mdc-theme-styles();
 }
 
+//
```
Comment (crisbeto) [inline src/material/chips/chip.scss:81]: Another leftover

File: src/material/core/tokens/m2/mdc/_chip.scss [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+  $is-dark: map.get($config, is-dark);
+  $surface: map.get(map.get($config, background), card);
+  $on-surface: if(_variable-safe-contrast-tone($surface, $is-dark) == 'dark', #000, #fff);
+  $standard-background: if(
+    meta.type-of($on-surface) == color and meta.type-of($surface) == color,
+    color.mix($on-surface, $surface, 12%),
+    $on-surface
+  );
+
+  $grey: if($is-dark, #fafafa, #212121);
```
Comment (crisbeto) [inline src/material/core/tokens/m2/mdc/_chip.scss:264]: We can get these colors from the `$grey-palette` (values 50 and 900).

**Review by andrewseguin — APPROVED** [review]

**Pull request merged**